flash/core/integrations/labelstudio/input.py

```diff
@@ -1,74 +1,153 @@
 import json
 import os
+from copy import deepcopy
+from dataclasses import dataclass
 from pathlib import Path
-from typing import Any, Mapping, Optional, Sequence, Union
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Set, Type
 
+import numpy as np
 import torch
 from pytorch_lightning.utilities.cloud_io import get_filesystem
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch.utils.data import Sampler
 
-from flash.core.data.auto_dataset import AutoDataset, IterableAutoDataset
-from flash.core.data.io.input import DataKeys, has_len, Input
+import flash
+from flash.core.data.io.input import DataKeys
+from flash.core.data.io.input_base import Input, IterableInput
+from flash.core.data.properties import ProcessState, Properties
 from flash.core.data.utils import image_default_loader
 from flash.core.utilities.imports import _PYTORCHVIDEO_AVAILABLE, _TEXT_AVAILABLE
 from flash.core.utilities.stages import RunningStage
 
+if _PYTORCHVIDEO_AVAILABLE:
+    from pytorchvideo.data.clip_sampling import make_clip_sampler
+
 if _TEXT_AVAILABLE:
     from transformers import AutoTokenizer
 
 
-class LabelStudioInput(Input):
-    """The ``LabelStudioInput`` expects the input to
-    :meth:`~flash.core.data.io.input.Input.load_data` to be a json export from label studio."""
+@dataclass(unsafe_hash=True, frozen=True)
+class LabelStudioState(ProcessState):
+    """The ``LabelStudioState`` stores the metadata loaded from the data."""
+
+    multi_label: bool
+    num_classes: Optional[int]
+    classes: Set
+    data_types: Set
+    tag_types: Set
+
+
+def _get_labels_from_sample(labels, classes):
+    """Translate string labels to int."""
+    sorted_labels = sorted(list(classes))
+    if isinstance(labels, list):
+        label = []
+        for item in labels:
+            label.append(sorted_labels.index(item))
+    else:
+        label = sorted_labels.index(labels)
+    return label
```

Comment:
Could one line this if you'd like:

```suggestion
    return [sorted_labels.index(item) for item in labels] if isinstance(labels, list) else sorted_labels.index(labels)
```

Reply:
Done !